tests/test_project.py

```diff
@@ -1288,30 +1292,49 @@ def test_script(self):
 
     def test_directives_hierarchy(self):
         project = self.mock_project()
+        cached_status = project._get_cached_status()
         for job in project:
             # Test submit JobOperations
-            job_ops = project._get_submission_operations(
-                (job,), project._get_default_directives(), names=["group2"]
+            job_ops = list(
+                project._get_submission_operations(
+                    aggregates=[(job,)],
```

Comment:
This change was from a bug in the tests. When `_get_submission_operations` is called, it requires a _sequence of aggregates_, not a _single aggregate_. We validate this in the public method that calls `_get_submission_operations`, but this test calls an internal private method. As a result, this test was being fed invalid data and was passing for the wrong reason. Somewhere (perhaps `_is_selected_aggregate`), a single job was being compared to an aggregate containing only that job, and that made the test pass because no job operations were generated. I only found this problem because of a mistake in the `Job.__eq__` method that was introduced in [PR 442](https://github.com/glotzerlab/signac/pull/442) is being fixed in [PR 455](https://github.com/glotzerlab/signac/pull/455). I have expanded the test to ensure that non-zero (submission/run) job operations are created and fixed the problem.